src/intern.rs

```diff
@@ -24,12 +39,28 @@ macro_rules! get_str {
 }
 
 impl InternedStr {
+    /// Return whether `self` is the empty string.
     pub fn is_empty(self) -> bool {
         self == *EMPTY_STRING
     }
+    /// Convert this identifier back into the original `String`, cloning it along the way.
+    ///
+    /// # See also
+    /// [`get_str!`](../macro.get_str.html)
+    ///
+    /// # Panics
+    /// This function will panic if another thread panicked while accessing the global string pool.
+    ///
     pub fn resolve_and_clone(self) -> String {
         get_str!(self).to_string()
```

Comment:
Not necessarily part of the change, but why is `.to_string()` used when they function is about cloning? They’re functionally the same, but it seems odd

Reply:
`get_str!` returns a `&str`, not a `String`.

Reply:
Ah, indeed it does, I always forget that clone on `&str` doesn’t do anything